### DISCUSSION TITLE
settings for reverse proxy

### DISCUSSION THREAD

**Discussion by balajeek** [photoprism/photoprism#1468] [General]
I setup photoprism on my qnap container and it runs fine.
I want to use the site on internal lan only, currently by http://192.168.1.9:2342/ and it works fine 

I use home assistant on https with traefik, i want to include this local site in to panel iframe so i could use only when i am home.  being HA as https it does not render a non-http site, so i have to make the photoprism site like https://<ha ip>/prism.

so i added traefik middleware toml but all is see is full page prism logo.  I am guessing there will be some setting i need to add to enable prism show under local self signed certificate.

any idea

**Comment by jared252016** [comment]
You probably need to enable websockets. I have not used traefik so unfortunately i cannot be of assistance, but with nginx proxy manager it works fine as long as websockets are enabled.

Get Outlook for Android<https://aka.ms/AAb9ysg>

________________________________
From: balajeek ***@***.***>
Sent: Wednesday, August 11, 2021 10:44:12 AM
To: photoprism/photoprism ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [photoprism/photoprism] settings for reverse proxy (#1468)


I setup photoprism on my qnap container and it runs fine.
I want to use the site on internal lan only, currently by http://192.168.1.9:2342/ and it works fine

I use home assistant on https with traefik, i want to include this local site in to panel iframe so i could use only when i am home. being HA as https it does not render a non-http site, so i have to make the photoprism site like https:///prism.

so i added traefik middleware toml but all is see is full page prism logo. I am guessing there will be some setting i need to add to enable prism show under local self signed certificate.

any idea

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/photoprism/photoprism/discussions/1468>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ASLNIDG5XPIS2MKZUZ2PUD3T4KLEZANCNFSM5B62R3CQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>.

**Reply by gitrahab** [reply]
> You probably need to enable websockets. I have not used traefik so unfortunately i cannot be of assistance, but with nginx proxy manager it works fine as long as websockets are enabled. Get Outlook for Android<https://aka.ms/AAb9ysg>

I have this same issue, but running from NGINX. I have a config below. What else is required to "enable websockets" as you describe?

```
        location /photoprism/ {
              proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
              proxy_set_header Host $host;
              proxy_pass http://localhost:2342/;
              proxy_ssl_server_name on;
              proxy_buffering off;
              proxy_http_version 1.1;
              proxy_set_header Upgrade $http_upgrade;
              proxy_set_header Connection "upgrade";
        }
```

**Comment by lastzero** [comment]
This example config comes pre-configured with HTTP using Traefik as reverse proxy: https://dl.photoprism.org/docker/cloud/

**Reply by balajeek** [reply]
my photoprism docker is running on a different machine than where my traefik is, so i guess i can't use labels on photoprism docker compose.  Instead i add a rules toml file for like i do for other sites (which are with domain names though), so i tried the same for photoprism and all i get is the big full page prism logo

here is my traefik toml rules for photoprism:
```
http.routers]
  [http.routers.prism-rtr]
      entryPoints = ["websecure"]
      rule = "PathPrefix(`/prism/`)"
      service = "prism-svc"
      [http.routers.prism-rtr.tls]
        certresolver = "myresolver"

[http.services]
  [http.services.prism-svc]
    [http.services.prism-svc.loadBalancer]
      passHostHeader = true
      [[http.services.prism-svc.loadBalancer.server.port=2342]]
        url = "http://192.168.1.7:2342"  # host's IP of photo prism 

```

**Comment by jared252016** [comment]
What web browser do you use? If you right click and choose inspect then go to console (tab) then reload the page, what error does it give?

Get Outlook for Android<https://aka.ms/AAb9ysg>

________________________________

**Reply by balajeek** [reply]
Error msg form console is:
The resource from “https://192.168.1.13/static/build/app.css?7a2bb49d” was blocked due to MIME type (“text/plain”) mismatch (X-Content-Type-Options: nosniff).  prism. [Lean more]

**Comment by jean-louis67** [comment]
Hello

I just encountered the same problem and after having struggled a lot, I found this solution very simple and logical:

My configuration of nginx is as follows:
```
location / photoprism {
  proxy_pass http://s-docker1.mydomain.eu:2342;
}
```
The other properties are strictly the same as those in the doc https://docs.photoprism.org/getting-started/proxies/nginx/

For this to work I just configured PhotoPrism as follows:

```
PHOTOPRISM_HTTP_PORT: 2342
PHOTOPRISM_SITE_URL: "http://s-docker.mydomain.eu/photoprism"
```
And it works as we want. 

What I understand is that the parameter _PHOTOPRISM_SITE_URL_ is used to constitute the hyperlinks found in the pages returned by PhotoPrism. We must therefore indicate to Photoprism in what context it is.
